### PR TITLE
rm deprecated 2.x api

### PR DESCRIPTION
**Pull request by aruniverse**
- Remove `ChangeSummaryManager.extractChangeSummaries` in core-backend
- Remove `handleInstancesPolymorphically` from `ContentInstancesOfSpecificClassesSpecification` in presentation-common
- updated tests in presentation-common and full-stack-tests

### CODE REVIEW

**Comment by aruniverse** [comment]
@diegopinate mind reviewing the changes to the ChangesetManager (in case it affects you with VC)

**Comment by diegopinate** [comment]
@aruniverse, it shouldn't impact it. We use the change summary API that's in the addon internally, and you didn't make any changes to the ChangedElementsManager, so this should be fine.

File: presentation/common/src/presentation-common/RulesetsFactory.ts [hunk 1]
```diff
@@ -224,7 +223,9 @@ const createComparison = (type: PrimitiveTypeDescription, name: string, operator
 
 const createMultiClassSpecification = (classInfo: Readonly<ClassInfo>): MultiSchemaClassesSpecification => {
   const [schemaName, className] = classInfo.name.split(":");
-  return { schemaName, classNames: [className] };
+  return classInfo.arePolymorphic !== undefined ?
+    { schemaName, classNames: [className], arePolymorphic: classInfo.arePolymorphic } :
+    { schemaName, classNames: [className] };
```
Comment (saskliutas) [inline presentation/common/src/presentation-common/RulesetsFactory.ts:228]:
```suggestion
  return { schemaName, classNames: [className], arePolymorphic: true }
```

File: presentation/common/src/presentation-common/EC.ts [hunk 2]
```diff
@@ -85,6 +85,7 @@ export interface ClassInfo {
   name: string;
   /** ECClass label */
   label: string;
+  arePolymorphic?: boolean;
```
Comment (saskliutas) [inline presentation/common/src/presentation-common/EC.ts:88]: Please remove new `arePolymorphic` property from `ClassInfo` types.

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: presentation/common/src/presentation-common/RulesetsFactory.ts (hunk 1 above)
Comment (Jake-Screen) [reply presentation/common/src/presentation-common/RulesetsFactory.ts]: change made

File: docs/changehistory/NextVersion.md [hunk 3]
```diff
@@ -298,6 +298,10 @@ In addition to upgrading iTwin.js core dependencies to `4.0`, there are some oth
 - Upgrade [iTwinUI](https://github.com/iTwin/iTwinUI) from v1 to v2.
 - `@itwin/presentation-backend`, `@itwin/presentation-common` and `@itwin/presentation-frontend` have new peer dependency `@itwin/ecschema-metadata`.
 
+### Removed deprecated items
```
Comment (pmconne) [inline docs/changehistory/NextVersion.md:301]: We already have a "Deprecated API removals" section under "Breaking changes".
Spelling: "polymorphicly" => "polymorphically".

**Review by saskliutas — APPROVED** [review]

File: docs/changehistory/NextVersion.md [hunk 4]
```diff
@@ -298,6 +302,10 @@ In addition to upgrading iTwin.js core dependencies to `4.0`, there are some oth
 - Upgrade [iTwinUI](https://github.com/iTwin/iTwinUI) from v1 to v2.
 - `@itwin/presentation-backend`, `@itwin/presentation-common` and `@itwin/presentation-frontend` have new peer dependency `@itwin/ecschema-metadata`.
 
+### ContentInstancesOfSpecificClassesSpecification
+
+The deprecated field `handleInstancesPolymorphically?` of [ContentInstancesOfSpecificClassesSpecification]($presentation-common) has been removed. To specify handling polymorphically, specify the value in `classes.arePolymorphic` or `excludedClasses.arePolymorphic`.
```
Comment (aruniverse) [inline docs/changehistory/NextVersion.md:307]:
```suggestion
The deprecated field `handleInstancesPolymorphically` of [ContentInstancesOfSpecificClassesSpecification]($presentation-common) has been removed. To specify handling polymorphically, specify the value in `classes.arePolymorphic` or `excludedClasses.arePolymorphic`.
```

**Pull request merged**